simpleviewinc/keg-herkin
# Zen 438 herkin UI

**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-438)

## Context

* Setting up the initial screens and modal for keg-herkin
* initially Display 3 component as modal ( ‘select test type’ dropdown (gherkin default), `Select test file` dropdown and `Create new file` btn)

## Goal

* Implement the modal with initial selectors to setup which test file user wants to see
* do some cleanup on some function/files

## Updates

* `repos/tap/actions/features/loadFeature.js`
    * will load in the feature file and changes to a specified screen if passed in 
* `repos/tap/actions/features/setFeatureActive.js`
    * cleanup
* `repos/tap/actions/init.js`
   * updated so it displays the init modal if no querystring is passed in
* `repos/tap/actions/modals/*`
   * basic modal actions, similar to the ones on evf
* `repos/tap/components/form/select/select.js`
   * custom select component using keg-components
* `repos/tap/components/modals/testSelectorModal.js`
   * the select modal component
   * user picks which tab, file and test type they want to initialize
* `repos/tap/components/runner/runner.js`
  * removing some unused lines 
* `repos/tap/hooks/useFeature.js`
  * minor adjust to be able to fetch feature based on the feature name
## Testing

### Setup
* Run `keg herkin pack run package=keg-herkin:zen-438-herkin-ui ports=5005:5005`
* navigate to http://herkin-zen-438-herkin-ui.local.kegdev.xyz/
* Verify that it loads the settings modal initially
  * [image](https://user-images.githubusercontent.com/3317835/107095060-5fe51900-67c5-11eb-9e55-60f09250a8e8.png)
  * 
### Google Search option
* Verify the `Select initial tab:` select contains 3 items: Editor/Builder/Runner. with `Editor` as default
* Verify `Test Type` is only `Gherkin`
* Verify `Select Test File` defaults to `Create new file`
* select `Google Search` as the test file
* click `Start`
* Verify that it navigates to the correct `Tab` and loads the `Google Search` Feature file

### Create new file option
* Refresh the browser to see the Settings modal
* Select `Create New File` 
* Select `Initial Tab`: Runner
* It should console log a placeholder message
* as a placeholder It should drop you in the `Test Runner` tab with the example jest test
